### DISCUSSION TITLE
How to change parameters of segmentation model in diarization pipeline

### DISCUSSION THREAD

**Discussion by pmbaumgartner** [pyannote/pyannote-audio#1105] [Q&A] [answered]
I'm really enjoying how easy it is to do diarization with the pipeline in `pyannote`, thanks so much for this awesome library. Currently I'm trying to do diarization on a meeting recording with 4 people, and the default pipeline is creating a lot of small segments where I know the speaker doesn't change. Currently I'm doing some post-processing steps on the output segments from the model, but I actually think it might be more productive to solve this problem using a parameter like `min_duration_off` in the segmentation model in the pipeline.

I couldn't find any examples on how to change these parameters in a pipeline. Is this something that's easily doable or do I essentially have to recreate the pipeline? Ideally I'm looking for something similar to the snippet below.

```python
from pyannote.audio import Pipeline
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization")

# change things like `min_duration_off` for the segmentation part of the pipeline

diarization = pipeline(AUDIO_FILE, num_speakers=4)
```

**Comment by hbredin** [comment, marked as answer]
`pipeline.segmentation.min_duration_off = FILL_GAPS_SHORTER_THAN_THAT_MANY_SECONDS` should do the trick.

**Reply by pmbaumgartner** [reply]
I get `AttributeError: 'str' object has no attribute 'min_duration_off'` with that. 

However, `pipeline._segmentation` (on the private attribute) doesn't error, but I think that's an instance of `pyannote.audio.core.inference.Inference` so it's just setting those attributes without them being used.

**Reply by hbredin** [reply]
Can you please let me know about your version of pyannote.audio?

**Reply by pmbaumgartner** [reply]
```bash
$  pip freeze | grep 'pyannote.audio'
pyannote.audio==2.0.1
```

**Reply by hbredin** [reply]
My bad. This won't work. You'd have to go with

```python
diarization = pipeline("audio.wav")
updated_diarization = diarization.support(collar=FILL_GAPS_SHORTER_THAN_THAT_MANY_SECONDS)
```

See [`Annotation.support` documentation](http://pyannote.github.io/pyannote-core/reference.html#pyannote.core.Annotation.support) .
